File: daily/2025/2918_Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long minSum(vector<int>& nums1, vector<int>& nums2) {
        long long s1 = 0;
        long long s2 = 0;
        bool nums1_contain_zero = false;
        bool nums2_contain_zero = false;
        for (int i : nums1){
            if (i == 0){
                nums1_contain_zero = true;
                s1++;
            }
            else s1 += i;
        }
        for (int i : nums2){
            if (i == 0){
                nums2_contain_zero = true;
                s2++;
            }
            else s2 += i;
        }
        if ((!nums1_contain_zero && s1 < s2) || (!nums2_contain_zero && s2 < s1)) return -1;
        return max(s1, s2);
    }
};

int main(){
    vector<int> nums1 = {3,2,0,1,0};
    vector<int> nums2 = {6,5,0};
    // vector<int> nums1 = {2,0,2,0};
    // vector<int> nums2 = {1,4};
    // vector<int> nums1 = {1,2,3,2};
    // vector<int> nums2 = {1,4,3};
    Solution S;
    cout << S.minSum(nums1, nums2) << endl;
    return 0;
}

File: daily_python/2918_Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.py
import os
import sys
from typing import List

class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        s1, s2, zero1, zero2 = 0, 0, 0, 0
        for i in nums1:
            s1 += i
            if i == 0:
                zero1 = True
                s1 += 1
        for i in nums2:
            s2 += i
            if i == 0:
                zero2 = True
                s2 += 1
        if (not zero1 and s1 < s2) or (not zero2 and s2 < s1):
            return -1
        return max(s1, s2)
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums1 = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            nums2 = list(map(int, lines[1].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((nums1, nums2))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums1, nums2 in dataset:
        results.append(solution.minSum(nums1, nums2))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')